machine.go

```diff
@@ -934,6 +934,27 @@ func (m *Machine) GetMetadata(ctx context.Context, v interface{}) error {
 	return nil
 }
 
+// DescribeInstanceInfo gets the VM info and unmarshals it into v
+func (m *Machine) DescribeInstanceInfo(ctx context.Context, v interface{}) error {
```

Comment:
Does it have to take `v interface{}`? The type of the value is fixed, I believe.

Reply:
For these API calls, using v interface{} would give the flexibility for the value to be not tied to one fixed type. I see that, this is the pattern used in rest of the API calls as well.

Reply:
We only have GetMetadata() as a getter and that confuses everyone apparently :)

So, GetMetadata() is the wrapper of [MMDS](https://github.com/firecracker-microvm/firecracker/blob/main/docs/mmds/mmds-user-guide.md), which can have any arbitrary data.

DescribeInstanceInfo is different since it is wrapping [this API](https://github.com/firecracker-microvm/firecracker/blob/6292f3289a8a5acdbd15d7e042a89ece95c31fb4/src/api_server/swagger/firecracker.yaml#L800), where we already know the type.

So you should return `(some data, error)` rather than returning `interface{}`. Go is a statically-typed language and types give developers hints about how to consume information correctly.

Reply:
I had the same problem before, and tried to keep consistent with GetMetadata(). @alakesh You may find some useful information from [GetBalloonConfig()](https://github.com/RoyceDavison/firecracker-go-sdk/blob/3e004c7509e3c21a6d772673d06a2eafee7eb8ac/machine.go#L1090)

Reply:
Thanks @RoyceDavison  @kzys, only worry is having inconsistency in the way the APIs are implemented. Updated the PR to use fixed type.